fwprovider/nodes/resource_acme_certificate.go

```diff
@@ -0,0 +1,882 @@
+							Description: "An optional alias domain for DNS validation. " +
+								"This allows you to validate the domain using a different domain's DNS records.",
+							Optional: true,
+						},
+					},
+				},
+			},
+			"force": schema.BoolAttribute{
+				Description: "Force certificate renewal even if the certificate is not due for renewal yet. " +
+					"Setting this to true will trigger a new certificate order on every apply.",
+				Optional: true,
+				Computed: true,
+				Default:  booldefault.StaticBool(false),
+			},
+			"certificate": schema.StringAttribute{
+				Description: "The PEM-encoded certificate data.",
+				Computed:    true,
+			},
+			"fingerprint": schema.StringAttribute{
+				Description: "The certificate fingerprint.",
+				Computed:    true,
+			},
+			"issuer": schema.StringAttribute{
+				Description: "The certificate issuer.",
+				Computed:    true,
+			},
+			"subject": schema.StringAttribute{
+				Description: "The certificate subject.",
+				Computed:    true,
+			},
+			"not_after": schema.StringAttribute{
+				Description: "The certificate expiration timestamp.",
+				Computed:    true,
+			},
+			"not_before": schema.StringAttribute{
+				Description: "The certificate start timestamp.",
+				Computed:    true,
+			},
+			"subject_alternative_names": schema.ListAttribute{
+				Description: "The certificate subject alternative names (SANs).",
+				Computed:    true,
+				ElementType: types.StringType,
+			},
+		},
+	}
+}
+
+// Configure adds the provider-configured client to the resource.
+func (r *acmeCertificateResource) Configure(
+	_ context.Context,
+	req resource.ConfigureRequest,
+	resp *resource.ConfigureResponse,
+) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	cfg, ok := req.ProviderData.(config.Resource)
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf("Expected config.Resource, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+
+		return
+	}
+
+	r.client = cfg.Client
+}
+
+// waitForCertificateAvailable polls ListCertificates until a certificate is available.
+// This replaces the fixed time.Sleep with a more robust retry mechanism.
+func (r *acmeCertificateResource) waitForCertificateAvailable(
+	ctx context.Context,
+	nodeClient *nodes.Client,
+) (*[]nodes.CertificateListResponseData, error) {
+	var certificates *[]nodes.CertificateListResponseData
+
+	err := retry.Do(
+		func() error {
+			certs, err := nodeClient.ListCertificates(ctx)
+			if err != nil {
+				return err
+			}
+
+			// Check if any certificates are found
+			if certs == nil || len(*certs) == 0 {
+				return fmt.Errorf("no certificates found yet")
+			}
+
+			certificates = certs
+
+			return nil
+		},
+		retry.Attempts(30),                    // Maximum 30 attempts
+		retry.Delay(1*time.Second),            // Start with 1 second delay
+		retry.DelayType(retry.BackOffDelay),   // Use exponential backoff
```

Comment:
exponential backoff with 30 attempts is a way too long IMO,
prob. just a fixed delay (5 sec? 10 sec?) backoff?